src/xdocs/checks/misc/leadingasteriskalign.xml.template

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document xmlns="http://maven.apache.org/XDOC/2.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/XDOC/2.0 https://maven.apache.org/xsd/xdoc-2.0.xsd">
+  <head>
+    <title>LeadingAsteriskAlign</title>
+  </head>
+  <body>
+    <section name="LeadingAsteriskAlign">
+      <p>Since Checkstyle 10.15</p>
+      <subsection name="Description" id="Description">
+        <p>
+          Checks alignment of leading asterisks in block comments and javadocs.
+          If asterisks is not aligned properly or
+          if there are consecutive asterisks then it will produce violation.
+          If no asterisks is used then that line will be ignored.
+        </p>
+      </subsection>
+
+      <subsection name="Properties" id="Properties">
+        <div class="wrapper">
+          <macro name="properties">
+            <param name="modulePath"
+                   value="src/main/java/com/puppycrawl/tools/checkstyle/checks/LeadingAsteriskAlignCheck.java"/>
+          </macro>
+        </div>
+      </subsection>
+
+      <subsection name="Examples" id="Examples">
+        <p id="Example1-config">
+          To configure the check:
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/leadingasteriskalign/Example1.java"/>
+          <param name="type" value="config"/>
+        </macro>
+        <p id="Example1-code">
+          Example:
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/leadingasteriskalign/Example1.java"/>
+          <param name="type" value="code"/>
+        </macro>
+        <p id="Example2-config">
+          To configure the check for asterisk to be aligned in left side.
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/leadingasteriskalign/Example2.java"/>
+          <param name="type" value="config"/>
+        </macro>
+        <p id="Example2-code">
+          Example:
+        </p>
+        <macro name="example">
+          <param name="path"
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/leadingasteriskalign/Example2.java"/>
+          <param name="type" value="code"/>
+        </macro>
+      </subsection>
+
+      <subsection name="Example of Usage" id="Example_of_Usage">
+        <ul>
+          <li>
+            <a href="https://github.com/search?q=path%3Aconfig%20path%3A**%2Fcheckstyle-checks.xml+repo%3Acheckstyle%2Fcheckstyle+LeadingAsteriskAlign">
+            Checkstyle Style</a>
+          </li>
```

Comment:
I've removed the `Sun Style` usage from here as we're currently not using this check in `sun_style.xml`, `mvn clean verify` was also complaining about it. As discussed here https://github.com/checkstyle/checkstyle/issues/6723#issuecomment-495578452 if we're going to add usage of this Module to our Sun Style Guide then we can open an separate issue in the future once the implementation of this check is done. @romani